### DISCUSSION TITLE
SignalR: WebSockets reconnect to scaled server.

### DISCUSSION THREAD

**Discussion by meareindel** [dotnet/aspnetcore#23142] [General] [answered]
There are docs at https://docs.microsoft.com/ru-ru/aspnet/core/signalr/scale?view=aspnetcore-3.1#redis-backplane describing requirements for sticky sessions, particularly this paragraph:

> Sticky sessions, also known as client affinity, is required, except when both of the following are true:
> All clients are configured to only use WebSockets.
> The SkipNegotiation setting is enabled in the client configuration. Once a connection is initiated on a server, the connection has to stay on that server.

Could you please elaborate how websocket reconnection is supposed to happen in that case? I mean, if I unplug network cable and plug back over time, how would websocket reconnect to the very same server from which it disconnected later if no sticky sessions are configured?

**Comment by davidfowl** [comment, marked as answer]
Reconnection spins up a new connection And connection id to potentially another server (depending on how the affinity was done) . There are no guarantees that you will hit the same server.

**Comment by meareindel** [comment]
Thanks! So in fact one always needs some kind of ARR configured for reconnection. I think it would be very helpful if mentioned in docs.
